packages/core/cache/nuxt/index.js

```diff
@@ -1,93 +1,117 @@
-  const invalidatorsRaw = options.server.invalidators || []
-
-  return invalidatorsRaw.map(invalidator => {
-    if (typeof invalidator === 'string') {
-      return require(path.resolve(invalidator)).default;
+/**
+ * Loads driver using path provided in the configuration.
+ */
+function createDriver (driver) {
+  const resolveDriverPath = name => {
+    if (isAbsolute(name) || name.startsWith('.')) {
+      return resolve(process.cwd(), name);
     }
+    
+    return require.resolve(driver[0], { paths: [ process.cwd() ] });
+  };
 
-    return invalidator
-  })
+  return Array.isArray(driver)
+    ? require(resolveDriverPath(driver[0])).default(driver[1])
+    : require(resolveDriverPath(driver)).default({});
 }
 
-const createDriver = (options) => {
-  const { driver } = options.server
-  const invalidators = createInvalidators(options)
-  if (Array.isArray(driver)) {
-    return require(path.resolve(driver[0])).default(driver[1], invalidators);
-  }
-
-  return require(path.resolve(driver)).default({}, invalidators);
-}
-
-function cacheModule (options) {
-  Logger.info('Installed cache');
+/**
+ * Main VSF cache module.
+ */
+export default function cacheModule (options) {
+  Logger.info('Installed Vue Storefront Cache plugin');
 
   // This part must be before the condition below
-  const resolvedDriver = createDriver(options)
   this.addPlugin({
-    src: path.resolve(__dirname, './plugin.js'),
+    src: resolve(__dirname, './plugin.js'),
     mode: 'server',
-    options: {
-      ...options.server,
-      driver: resolvedDriver
-    }
+    options
   });
 
   if (!this.nuxt || !this.nuxt.renderer) {
-    return
+    return;
   }
-  const createRenderer = rendererFactory.bind(this)
-
-  if (options.server.invalidateEndpoint) {
-    const { middleware, extend } = createMiddleware({});
 
-    extend((app) => {
-      app.get(options.server.invalidateEndpoint, async (req, res) => {
-        await resolvedDriver.invalidate({ req, res });
+  // Create cache driver
+  const driver = createDriver(options.driver)
 
-        res.send()
-      })
-    })
+  // Create invalidation endpoint if necessary
+  createInvalidationEndpoint.call(this, driver, options.invalidation);
 
-    this.addServerMiddleware(middleware);
-  }
-
-  createRenderer(async (route, context, render) => {
+  // Create renderer
+  createRenderer.call(this, async (route, context, render) => {
     const getTags = () => {
-      if (context.req.$vsfCache && context.req.$vsfCache.tagsSet) {
-        return Array.from(context.req.$vsfCache.tagsSet)
-      }
-
-      return []
+      const tags = context.req.$vsfCache && context.req.$vsfCache.tagsSet
+        ? Array.from(context.req.$vsfCache.tagsSet)
+        : [];
+      
+      return tags.map(({ prefix, value }) => `${prefix}${value}`);
     }
 
     try {
-      return await resolvedDriver.invoke({ route, context, getTags, render });
+      return await driver.invoke({
+        key: `page:${ route }`,
```

Comment:
you already has access to route in the driver so that field is useless here - decide in the driver what and how you want to store pages